utilities/cli-runner/src/main/kotlin/org/jetbrains/kotlin/cli/utilities/main.kt

```diff
@@ -40,5 +42,17 @@ private fun mainImpl(args: Array<String>, konancMain: (Array<String>) -> Unit) {
 
 fun main(args: Array<String>) = mainImpl(args, ::konancMain)
 
-fun daemonMain(args: Array<String>) = mainImpl(args, ::konancMainForGradle)
+private fun setupClangEnv() {
+    setEnv("LIBCLANG_NOTHREADS", "1")
+    setEnv("LIBCLANG_DISABLE_CRASH_RECOVERY", "1")
+}
+
+fun daemonMain(args: Array<String>) {
+    setupClangEnv()
+    try {
+        mainImpl(args, ::konancMainForGradle)
```

Comment:
One major problem with running cinterop in daemon might be:
it prints messages to stdout/stderr, and these won't be displayed to user. (That's why we use different compiler entry points for CLI and daemon). Please make sure that cinterop and libclang don't do that.